webpack-app/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "webpack-app",
+  "version": "1.0.0",
+  "description": "This is my first webpack application",
+  "main": "index.js",
+  "scripts": {
+    "start::dev": "webpack serve --config ./webpack.config.dev.js",
```

Comment:
with [cross-env](https://www.npmjs.com/package/cross-env) you can use it, like this:
```
"start::dev": "webpack serve",
"build::prod": "cross-env NODE_ENV=production webpack",
```

And because of the condition in your `webpack.config.js` it will use right config file